src/plugins/ui_stateful/clipboard.ts

```diff
@@ -518,36 +524,35 @@ export class ClipboardPlugin extends UIPlugin {
   }
 
   private getHTMLContent(): string {
-    if (!this.copiedData?.cells) {
-      return `<div data-clipboard-id="${this.clipboardId}">\t</div>`;
-    }
-    const cells = this.copiedData.cells;
-    if (cells.length === 1 && cells[0].length === 1) {
-      return `<div data-clipboard-id="${this.clipboardId}">${this.getters.getCellText(
-        cells[0][0].position
-      )}</div>`;
-    }
-    if (!cells[0][0]) {
+    let innerHTML: string = "";
+    const cells = this.copiedData?.cells;
+    if (!cells) {
+      innerHTML = "\t";
+    } else if (cells.length === 1 && cells[0].length === 1) {
+      innerHTML = `${this.getters.getCellText(cells[0][0].position)}`;
+    } else if (!cells[0][0]) {
       return "";
-    }
-
-    let htmlTable = `<div data-clipboard-id="${this.clipboardId}"><table border="1" style="border-collapse:collapse">`;
-    for (const row of cells) {
-      htmlTable += "<tr>";
-      for (const cell of row) {
-        if (!cell) {
-          continue;
+    } else {
+      let htmlTable = `<table border="1" style="border-collapse:collapse">`;
+      for (const row of cells) {
+        htmlTable += "<tr>";
+        for (const cell of row) {
+          if (!cell) {
+            continue;
+          }
+          const cssStyle = cssPropertiesToCss(
+            cellStyleToCss(this.getters.getCellComputedStyle(cell.position))
+          );
+          const cellText = this.getters.getCellText(cell.position);
+          htmlTable += `<td style="${cssStyle}">` + xmlEscape(cellText) + "</td>";
         }
-        const cssStyle = cssPropertiesToCss(
-          cellStyleToCss(this.getters.getCellComputedStyle(cell.position))
-        );
-        const cellText = this.getters.getCellText(cell.position);
-        htmlTable += `<td style="${cssStyle}">` + xmlEscape(cellText) + "</td>";
+        htmlTable += "</tr>";
       }
-      htmlTable += "</tr>";
+      htmlTable += "</table>";
+      innerHTML = htmlTable;
     }
-    htmlTable += "</table></div>";
-    return htmlTable;
+    const serializedData = JSON.stringify(this.getgridData());
+    return `<div data-osheet-clipboard='${serializedData}'>${innerHTML}</div>`;
```

Comment:
I'm 95% sure this will break the HTML. Amprsand, quotes and other stuff need to be escaped at the very least